src/megaapi_impl.cpp

```diff
@@ -14216,73 +14278,155 @@ void MegaApiImpl::getua_result(byte* data, unsigned len, attr_t type)
 
 void MegaApiImpl::getua_result(TLVstore *tlv, attr_t)
 {
-    if(requestMap.find(client->restag) == requestMap.end()) return;
-    MegaRequestPrivate* request = requestMap.at(client->restag);
-    if(!request || (request->getType() != MegaRequest::TYPE_GET_ATTR_USER)) return;
+    error e = API_OK;
+    MegaRequestPrivate* request = NULL;
+    auto it = requestMap.find(client->restag);
+    if (it == requestMap.end() || !(request = it->second)
+           || (request->getType() != MegaRequest::TYPE_GET_ATTR_USER && request->getType() != MegaRequest::TYPE_SET_ATTR_USER)
+           || (request->getType() == MegaRequest::TYPE_SET_ATTR_USER && request->getParamType() != MegaApi::USER_ATTR_ALIAS)) return;
 
     if (tlv)
     {
-        // TLV data usually includes byte arrays with zeros in the middle, so values
-        // must be converted into Base64 strings to avoid problems
-        MegaStringMap *stringMap = new MegaStringMapPrivate(tlv->getMap(), true);
-        request->setMegaStringMap(stringMap);
-
-        switch (request->getParamType())
+        if (request->getType() == MegaRequest::TYPE_SET_ATTR_USER)
         {
-            // prepare request params to know if a warning should show or not
-            case MegaApi::USER_ATTR_RICH_PREVIEWS:
+            bool modified = false;
+            const char *key, *alias, *rcvAlias;
+            MegaStringMap *newAliasesMap = request->getMegaStringMap();
+            MegaStringList *keys = newAliasesMap->getKeys();
+
+            // Iterate though the string map <uh_B64, alias>
+            for (int i=0; i < keys->size(); i++)
             {
-                const char *num = stringMap->get("num");
+                key = keys->get(i);
+                alias = newAliasesMap->get(key);
 
-                if (request->getNumDetails() == 0)  // used to check if rich-links are enabled
+                // If the user alias doesn't exists or has been changed, update the TLV container
+                if (alias)
```

Comment:
Is it possible to get an `alias` set to `null`? :thinking:

Reply:
MegaStringMapPrivate::get returns NULL if don't find the key